pkg/controller/dbsyncer/generic_db_to_transport_syncer.go

```diff
@@ -81,28 +76,70 @@ func (syncer *genericDBToTransportSyncer) initLastUpdateTimestampFromTransport()
 }
 
 func (syncer *genericDBToTransportSyncer) syncBundle(ctx context.Context) {
```

Comment:
as suggested in previous comments - I would leave original syncBundle function as is.
you can add error as a return value which might help you maintain the interval policy functions easily.
then you can implement the very straight forward periodicSync function:
 ```
 func (syncer *genericDBToTransportSyncer) periodicSync(ctx context.Context) {
	ticker := time.NewTicker(syncer.intervalPolicy.GetInterval())

	for {
		select {
		case <-ctx.Done(): // we have received a signal to stop
			ticker.Stop()
			return

		case <-ticker.C:
			if err := syncer.syncBundle(ctx); err != nil {
				// log error
				// call interval policy appropriate func
			} else {
				// call interval policy appropriate func 
			}
			syncer.reevaluateInterval()
		}
	}
}
 ```

Reply:
done